### ISSUE TITLE
Null Pointer exception when picking image from gallery (on Genymotion Nexus 7)

### ISSUE THREAD

**Issue by vinaywadhwa** [ParkSangGwon/TedBottomPicker#15]
Caused by: java.lang.NullPointerException: Attempt to read from field 'int gun0912.tedbottompicker.TedBottomPicker$Builder.peekHeight' on a null object reference
                                                                   at gun0912.tedbottompicker.TedBottomPicker.setupDialog(TedBottomPicker.java:102)
                                                                   at android.support.v4.app.DialogFragment.getLayoutInflater(DialogFragment.java:310)
                                                                   at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1108)
                                                                   at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1290)
                                                                   at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1272)
                                                                   at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2149)
                                                                   at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:201)
                                                                   at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:600)
                                                                   at android.support.v7.app.AppCompatActivity.onStart(AppCompatActivity.java:178)
                                                                   at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1237)
                                                                   at android.app.Activity.performStart(Activity.java:6253)
                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2379)
                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
                                                                   at android.app.ActivityThread.-wrap11(ActivityThread.java) 
                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) 
                                                                   at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                   at android.os.Looper.loop(Looper.java:148) 
                                                                   at android.app.ActivityThread.main(ActivityThread.java:5417)

**Comment by ivandiazmolina** [comment]
Same problem in library's code inside method "setupDialog"

if(builder.peekHeight>0){
        // ((BottomSheetBehavior) behavior).setPeekHeight(1500);
        ((BottomSheetBehavior) behavior).setPeekHeight(builder.peekHeight);
}

**Comment by ParkSangGwon** [comment]
That is little strange.
That error log told us `behavior is null`.
But I alway null check when use behavior.
This is source code.
```javascript
  if (behavior != null && behavior instanceof BottomSheetBehavior) {
            ((BottomSheetBehavior) behavior).setBottomSheetCallback(mBottomSheetBehaviorCallback);
            if(builder.peekHeight>0){
                // ((BottomSheetBehavior) behavior).setPeekHeight(1500);
                ((BottomSheetBehavior) behavior).setPeekHeight(builder.peekHeight);
            }

        }
`

You can see ` if (behavior != null && behavior instanceof BottomSheetBehavior)`.
So behavior can not be null.
Hm... 
Did you have always error?

**Comment by ivandiazmolina** [comment]
Yes, but variable is null because fragment dialog is rebuilt after selecting the image

**Comment by exloong** [comment]
Same error to me, happens every time when you confirm selecting the photo you just took

**Comment by kushanshah11** [comment]
I have same error occurred in Samsung Galaxy S4 while i taking picture from camera.

guys can you please solve it as soon as possible?


Thanks in Advance.

**Comment by kushanshah11** [comment]
@ParkSangGwon  can you please Look into this issue ???

anyone solved this issue please post Resolution for this issue.


Thanks in advance.

**Comment by ParkSangGwon** [comment]
@vinaywadhwa @ivandiazmolina @exloong @kushanshah11 
That error occur by rotation.
I fix rotation error at 1.0.6 version.
Now you can use this library without `NullPointerException`

**Comment by kushanshah11** [comment]
Hi @ParkSangGwon  ,

                                                                 Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=65537, result=-1, data=null} to activity {com.iotconnect/com.iotconnect.activity.DrawerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.getPath()' on a null object reference
                                                                    at android.app.ActivityThread.deliverResults(ActivityThread.java:4067)
                                                                    at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3389)
                                                                    at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3434) 
                                                                    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2772) 
                                                                    at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4471) 
                                                                    at android.app.ActivityThread.access$1000(ActivityThread.java:177) 
                                                                    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1455) 
                                                                    at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                    at android.os.Looper.loop(Looper.java:145) 
                                                                    at android.app.ActivityThread.main(ActivityThread.java:5951) 
                                                                    at java.lang.reflect.Method.invoke(Native Method) 
                                                                    at java.lang.reflect.Method.invoke(Method.java:372) 
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1388) 
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1183) 
                                                                 Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.getPath()' on a null object reference
                                                                    at gun0912.tedbottompicker.TedBottomPicker.onActivityResultCamera(TedBottomPicker.java:467)
                                                                    at gun0912.tedbottompicker.TedBottomPicker.onActivityResult(TedBottomPicker.java:453)
                                                                    at android.support.v4.app.FragmentActivity.onActivityResult(FragmentActivity.java:164)
                                                                    at android.app.Activity.dispatchActivityResult(Activity.java:6549)
                                                                    at android.app.ActivityThread.deliverResults(ActivityThread.java:4063)
                                                                    at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3389) 
                                                                    at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3434) 
                                                                    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2772) 
                                                                    at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4471) 
                                                                    at android.app.ActivityThread.access$1000(ActivityThread.java:177) 
                                                                    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1455) 
                                                                    at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                    at android.os.Looper.loop(Looper.java:145) 
                                                                    at android.app.ActivityThread.main(ActivityThread.java:5951) 
                                                                    at java.lang.reflect.Method.invoke(Native Method) 
                                                                    at java.lang.reflect.Method.invoke(Method.java:372) 
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1388) 
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1183) 

i m still facing same issue while i capture pic from SAMSUNG S4.

and yes i have used latest version 1.0.6 

can you please Look into this issue.

Thanks in advance
